packages/react-composites/src/composites/CallComposite/utils/Utils.ts

```diff
@@ -184,3 +186,70 @@ export const computeVariant = (
     return 'NO_STATE';
   }
 };
+
+/**
+ * Hook to override the avatar persona data of the users
+ * - Calls {@link AvatarPersonaDataCallback} on each render for provided `userIds`
+ * - Returns an array of the same length as `userIds`. Entries in the array may be undefined if there is no data to return.
+ * @private
+ */
+export const useCustomAvatarPersonaData = (
+  userIds: (string | undefined)[],
+  callBack?: AvatarPersonaDataCallback
+): (AvatarPersonaData | undefined)[] => {
+  const [data, setData] = React.useState<(AvatarPersonaData | undefined)[]>([]);
+  useEffect(() => {
+    (async () => {
+      if (!callBack) {
+        const undefinedData: undefined[] = [];
+        for (let i = 0; i < userIds.length; i++) {
+          undefinedData.push(undefined);
+        }
+        setData(undefinedData);
```

Comment:
This one not a nit.

This sets a new array on each render when `callback` is `undefined`. Instead, similar to the non-default case, we should return the same array each time so that the component that uses this hook doesn't render again.

My suggestion: `shouldUpdate` already handles when elements in the array are `undefined`. Just use that here as well before calling `setData`.